### ISSUE TITLE
Create Scribble Viewer (View not edit)

### ISSUE THREAD

**Issue by alexrabin** [fyzio/scribble#10]
So I had a use case where I needed to just view a scribble sketch. As I started working on this solution I realized that there was a lot of extra space in my sketch (ie. you draw a smiley face in the center of the canvas). So I came up with some helper functions that helped me create a scribble viewer that crops the sketch where it removes the whitespace above and below the main sketch.

These are the extensions I needed to help create the Sketch viewer.
```
extension SketchExt on Sketch {
  Sketch croppedSketch() {
    var newLines = List<SketchLine>.empty(growable: true);
    double minY = double.maxFinite;
    for (var line in lines) {
      var temp = line.points
          .reduce((value, element) => value.y < element.y ? value : element);
      if (temp.y < minY) {
        minY = temp.y;
      }
    }

    newLines = lines.map((e) {
      var newE = e;
      List<Point> points = List<Point>.empty(growable: true);
      for (var p in e.points) {
        var newP = p.copyWith(y: p.y - minY);
        points.add(newP);
      }
      newE = newE.copyWith(points: points);
      return newE;
    }).toList();
    return copyWith(lines: newLines);
  }

  double getTotalHeight() {
    double minY = double.maxFinite;
    double maxY = -double.maxFinite;
    for (var line in lines) {
      var minTemp = line.points
          .reduce((value, element) => value.y < element.y ? value : element);
      var maxTemp = line.points
          .reduce((value, element) => value.y > element.y ? value : element);
      if (minTemp.y < minY) {
        minY = minTemp.y;
      }
      if (maxTemp.y > maxY) {
        maxY = maxTemp.y;
      }
    }
    return maxY - minY;
  }
}
```

Here is the sketch viewer in motion
```
Widget sketchViewer(Sketch sketch){
    return SizedBox(
        height: sketch
            .getTotalHeight(),
        child: Scribble(
          notifier: ScribbleNotifier(
              allowedPointersMode: ScribblePointerMode.mouseOnly,
              sketch: sketch
                  .croppedSketch()),
          drawPen: true,
        ),
      );
}
```

I figured this could help somebody. I don't know if it is worth creating a pr for this or not.

**Comment by timcreatedit** [comment]
I like the idea of a ScribbleViewer widget, as well as being able to crop the sketch or at least get it's bounding box! I'm not quite sure yet which implementation would fit the package best as I definitely want to keep everything as unopinionated as possible, but I'll look into it! Thanks for taking the time

**Comment by timcreatedit** [comment]
This has been implemented in 0.3.0 with the ``ScribbleSketch`` widget :)
